### PR TITLE
New: Added progressive ItemsComponentModel bars

### LINKED ISSUE #220: Use progressive bars for ItemsComponentModel components
### Subject of the issue
Progress bars should represent the increment completion of Narrative, HotGraphic and Accordion.

Original comment: https://github.com/adaptlearning/adapt-contrib-pageLevelProgress/issues/181#issuecomment-1526028319

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes #220 

### Fix
* Allow bars to update when items are visited on presentation components with items

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by kirsty-hames — COMMENTED** [review]
This is a nice addition thanks @oliverfoster. I just have an observation regarding the Narrative component. Narrative is the only core presentation component that doesn't require an interaction for items to mark as complete. This means the first item is always complete even before the component is in view.

File: js/PageLevelProgressIndicatorView.js [hunk 1]
```diff
@@ -51,6 +52,12 @@ class PageLevelProgressIndicatorView extends Backbone.View {
   }
 
   calculatePercentage() {
+    const isPrensentationComponentWithItems = (!this.model.isTypeGroup('question') && this.model instanceof ItemsComponentModel);
```
Comment (swashbuck) [inline js/PageLevelProgressIndicatorView.js:55]: Should be `isPresentationComponentWithItems` not `isPrensentationComponentWithItems`.

File: js/PageLevelProgressIndicatorView.js [hunk 2]
```diff
@@ -51,6 +52,12 @@ class PageLevelProgressIndicatorView extends Backbone.View {
...
+    if (isPrensentationComponentWithItems) {
```
Comment (swashbuck) [inline js/PageLevelProgressIndicatorView.js:56]: Should be `isPresentationComponentWithItems` not `isPrensentationComponentWithItems`.

File: js/PageLevelProgressIndicatorView.js (hunk 1 above)
Comment (swashbuck) [inline js/PageLevelProgressIndicatorView.js:55]:
```suggestion
    const isPresentationComponentWithItems = (!this.model.isTypeGroup('question') && this.model instanceof ItemsComponentModel);
```

File: js/PageLevelProgressIndicatorView.js (hunk 2 above)
Comment (swashbuck) [inline js/PageLevelProgressIndicatorView.js:56]:
```suggestion
    if (isPresentationComponentWithItems) {
```

**Review by kirsty-hames — APPROVED** [review]

**Pull request merged**